### ISSUE TITLE
Train and evaluate without image, calib files

### ISSUE THREAD

**Issue by sunwonHong** [traveller59/second.pytorch#457]
It work well with kitti dataset Thank you!

but i want to train with my own dataset. so i changed my dataset to kitti format.

The dataset i have is including only point cloud data not images, calib.

Can i train and evaluate my dataset without calib and image files?

If possible, let me know how can i train that on simple way.

**Comment by wj-shin** [comment]
사진은 시각화를 위해 존재하지 학습을 하기 위해 존재하지 않습니다. 그래서 키티 데이터셋의 사진을 넣고 학습을 진행해도 상관이 없을 것 같습니다.. 
If you cannot read korean, tell me

**Comment by wj-shin** [comment]
학습을 성공하셨나요??

**Comment by sunwonHong** [comment]
Thank you for replying.

I am still trying. but When i use my pointcloud with kitti image and calib, i guess there are some of problem on that.

Firstly, Couldn't the points be removed because of remove_outside parts in the code create_data.py?
Next, it is just a question. If I use any images and calib for eval, is there no problem with accuracy at all?

**Comment by wj-shin** [comment]
haha... I have a mistake when I tag different issue.

You can refer to https://github.com/traveller59/second.pytorch/issues/340

In here you can read  "the camera data will only be used for indexing and visualization in the viewer"

I think "the camera data" means image and calib

and calib data also use when measure aos(one of indicator)

**Comment by young147** [comment]
Hi,I also got the same problems that train without image and clib files.When I use my own dataset with the fabricated calib file and label_2 files(It means just used a part of its infos,such as the location and dimensions from lidar measured data which cannot corresponding to camera data),it's impossible to use the trained model as a result of the AP was always zero with the config file of car.fhd.config,so,is there any way to train my own lidar data without the calib and image files?

**Comment by young147** [comment]
> it is just a question. If I use any images an

The remove_outside in create_data.py is aim to crop the point cloud in the camera vision.

**Comment by sunwonHong** [comment]
I just succeeded in evaluation for all.fhd.config :) But still training was not successful.

Anyway, i am not sure what you have.
Do you mean  you have appropriate point cloud, label, calib file? without images?
or appropriate point cloud, calib file? without images, lables?

**Comment by sunwonHong** [comment]
I think you need to fill image coordinates in label file even if it is not exactly. I copied image coordinates with kitti label file even if it is not exactly.
After change, when it still be an error, let me know what error did you have

**Comment by young147** [comment]
> I think you need to fill image coordinates in label file even if it is not exactly. I copied image coordinates with kitti label file even if it is not exactly.
> After change, when it still be an error, let me know what error did you have

Ok,I will try,thank you so much.

**Comment by young147** [comment]
> I think you need to fill image coordinates in label file even if it is not exactly. I copied image coordinates with kitti label file even if it is not exactly.
> After change, when it still be an error, let me know what error did you have

It has been proved that the image and image coordinates have no effect on training result.

**Comment by clytze0216** [comment]
@young147
Hi, I want to train with my own datasets without image and calib . But I met an error which is: "invalid configuration argument"
Did you succeed train the model without image and calib ?
Looking forward to your reply! Thanks a lot!

**Comment by young147** [comment]
> @young147
> Hi, I want to train with my own datasets without image and calib . But I met an error which is: "invalid configuration argument"
> Did you succeed train the model without image and calib ?
> Looking forward to your reply! Thanks a lot!

Hi,as for me,I only train succeed without "correct" image info,which means adequate quantity of "fake" image and correct calib files are essential for a training.

**Comment by clytze0216** [comment]
> > @young147
> > Hi, I want to train with my own datasets without image and calib . But I met an error which is: "invalid configuration argument"
> > Did you succeed train the model without image and calib ?
> > Looking forward to your reply! Thanks a lot!
> 
> Hi,as for me,I only train succeed without "correct" image info,which means adequate quantity of "fake" image and correct calib files are essential for a training.

Thanks for your reply!!  Did you use the calib data of KITTI or the calib make by yourself? If you use your own calib ,could you tell me how do you get it? Thanks agan!

**Comment by young147** [comment]
> > > @young147
> > > Hi, I want to train with my own datasets without image and calib . But I met an error which is: "invalid configuration argument"
> > > Did you succeed train the model without image and calib ?
> > > Looking forward to your reply! Thanks a lot!
> > 
> > 
> > Hi,as for me,I only train succeed without "correct" image info,which means adequate quantity of "fake" image and correct calib files are essential for a training.
> 
> Thanks for your reply!! Did you use the calib data of KITTI or the calib make by yourself? If you use your own calib ,could you tell me how do you get it? Thanks agan!

Not at all.It is better for you to make your own calib file according to the official kitti dataset documentation,firstly,you should get the intrinsic parameters of your camera,and get the transformation matrix of Trv2c，and fill the R0_rect with identity matrix if you only used one camera,and P1,P2 as well.That what I've done to train my own dataset,hope you make it.

**Comment by evil-master** [comment]
> > > > @young147
> > > > Hi, I want to train with my own datasets without image and calib . But I met an error which is: "invalid configuration argument"
> > > > Did you succeed train the model without image and calib ?
> > > > Looking forward to your reply! Thanks a lot!
> > > 
> > > 
> > > Hi,as for me,I only train succeed without "correct" image info,which means adequate quantity of "fake" image and correct calib files are essential for a training.
> > 
> > 
> > Thanks for your reply!! Did you use the calib data of KITTI or the calib make by yourself? If you use your own calib ,could you tell me how do you get it? Thanks agan!
> 
> Not at all.It is better for you to make your own calib file according to the official kitti dataset documentation,firstly,you should get the intrinsic parameters of your camera,and get the transformation matrix of Trv2c，and fill the R0_rect with identity matrix if you only used one camera,and P1,P2 as well.That what I've done to train my own dataset,hope you make it.

您好，我最近研究如何训练自己的数据集，遇到一些问题，想请教一下可以吗？方便的话，可以加我的qq 1097715139
